Mathlib/LinearAlgebra/Matrix/ToLinearEquiv.lean

```diff
@@ -196,4 +200,39 @@ alias nondegenerate_iff_det_ne_zero ↔ Nondegenerate.det_ne_zero Nondegenerate.
 
 end Nondegenerate
 
+end LinearEquiv
+
+section Determinant
+
+open BigOperators
+
+lemma det_ne_zero_of_neg [DecidableEq n] {S : Type _} [LinearOrderedCommRing S]
```

Comment:
Can you add a docstring explaining this lemma? Can you also add a variant that requires each column to have a positive sum instead of each column?

Reply:
Is this theorem a special case of this one? https://en.wikipedia.org/wiki/Diagonally_dominant_matrix ?

Reply:
Yes, it's a combination of diagonal dominance and the [Gershgorin circle theorem](https://en.wikipedia.org/wiki/Gershgorin_circle_theorem). If we don't already have the latter, I would prefer to see that instead / in addition.

Reply:
> Yes, it's a combination of diagonal dominance and the [Gershgorin circle theorem](https://en.wikipedia.org/wiki/Gershgorin_circle_theorem). If we don't already have the latter, I would prefer to see that instead / in addition.

Indeed. I don't see the Gershgorin theorem anywhere in Mathlib though. I'll see if it is easy to prove. If so, I'll add it and use it to prove this lemma. If not, I'll just add a note saying that this lemma can be proved using Gershgorin theorem once somebody adds it to Mathlib.

Reply:
Ok. Here is a new version proving Gershgorin's circle theorem and using it to deduce that strictly dominant diagonal matrices have nonzero determinant. The statement of the theorem feels a bit strange since there is no API for Eigenvalues of matrices (as far as I can tell). It should probably be changed once the API is added.

Reply:
Nice I made an attempt here, but I was hesitant because I used `NormedField` while your theorem only required `LinearCommOrderedRing`

https://github.com/leanprover-community/mathlib4/compare/MohanadAhmed/diagonally_dominant_matrix?expand=1

Reply:
> Nice I made an attempt here, but I was hesitant because I used `NormedField` while your theorem only required `LinearCommOrderedRing`. But from your implementation it iseems it does require normed field after all

I used `LinearCommOrderedRing` in the original proof since it was the most general case I could prove. In fact, I only need the case for ℝ. As you said, the proof of Gershgorin needs `NormedField` so that it is good enough for me but we cannot get from it the original statement in full generality.